--- asterisk-1.4.39.1/apps/app_waitforsilence.c	2010-01-13 09:16:12.000000000 -0800
+++ asterisk-1.4.39.1-vici/apps/app_waitforsilence.c	2011-02-08 18:15:06.000000000 -0800
@@ -83,6 +83,7 @@ static int do_waiting(struct ast_channel
 	int res = 0;
 	struct ast_dsp *sildet;	 /* silence detector dsp */
  	time_t now;
+	int loop_count = 1;
 
 	rfmt = chan->readformat; /* Set to linear mode */
 	res = ast_set_read_format(chan, AST_FORMAT_SLINEAR);
@@ -128,8 +129,19 @@ static int do_waiting(struct ast_channel
 			}
 		}
 
-		if (option_verbose > 6)
-			ast_verbose(VERBOSE_PREFIX_3 "Got %dms silence< %dms required\n", dspsilence, silencereqd);
+		loop_count++;
+
+		/* Lets not spam the logs */
+		if (loop_count >= 51) {
+			if ((option_verbose > 6) && (option_verbose <= 21))
+				ast_verbose(VERBOSE_PREFIX_3 "Got %dms silence < %dms required\n", dspsilence, silencereqd);
+			loop_count = 1;
+		}
+
+		/* They must really want us to spam the logs */
+		if (option_verbose > 21)
+			ast_verbose(VERBOSE_PREFIX_3 "Got %dms silence < %dms required\n", dspsilence, silencereqd);
+
 
 		if (dspsilence >= silencereqd) {
 			if (option_verbose > 2)
