--- asterisk-1.4.39.1/contrib/scripts/get_ilbc_source.sh	2009-04-22 07:29:28.000000000 -0700
+++ asterisk-1.4.39.1-vici/contrib/scripts/get_ilbc_source.sh	2013-04-28 05:14:16.000000000 -0700
@@ -22,7 +22,7 @@ read tmp
 
 wget -P codecs/ilbc http://www.ietf.org/rfc/rfc3951.txt
 
-wget -q -O - http://www.ilbcfreeware.org/documentation/extract-cfile.awk | tr -d '\r' > codecs/ilbc/extract-cfile.awk
+wget -q -O - http://www.ilbcfreeware.org/documentation/extract-cfile.txt | tr -d '\r' > codecs/ilbc/extract-cfile.awk
 
 (cd codecs/ilbc && awk -f extract-cfile.awk rfc3951.txt)
 
