--- asterisk-1.4.39.1/include/asterisk/channel.h	2010-11-22 10:46:26.000000000 -0800
+++ asterisk-1.4.39.1-vici/include/asterisk/channel.h	2011-02-08 18:16:22.000000000 -0800
@@ -645,6 +645,15 @@ enum {
 	 * instead of actually hanging up.
 	 */
 	AST_SOFTHANGUP_UNBRIDGE =  (1 << 6),
+
+
+	/*!
+	 * \brief All softhangup flags.
+	 *
+	 * This can be used as an argument to ast_channel_softhangup_clear
+	 * to clear all softhangup flags from a channel.
+	 */
+	AST_SOFTHANGUP_ALL =       (0xFFFFFFFF)
 };
 
 
@@ -834,6 +843,20 @@ int ast_softhangup(struct ast_channel *c
  * \param reason an AST_SOFTHANGUP_* reason code */
 int ast_softhangup_nolock(struct ast_channel *chan, int cause);
 
+/*!
+ * \brief Clear a set of softhangup flags from a channel
+ *
+ * Never clear a softhangup flag from a channel directly.  Instead,
+ * use this function.  This ensures that all aspects of the softhangup
+ * process are aborted.
+ *
+ * \param chan the channel to clear the flag on
+ * \param flag the flag or flags to clear
+ *
+ * \return Nothing.
+ */
+void ast_channel_clear_softhangup(struct ast_channel *chan, int flag);
+
 /*! \brief Check to see if a channel is needing hang up 
  * \param chan channel on which to check for hang up
  * This function determines if the channel is being requested to be hung up.
