--- MaxScale-1.4.0.orig/server/modules/filter/slavelag.c	2016-03-17 12:38:50.804598574 +0100
+++ MaxScale-1.4.0/server/modules/filter/slavelag.c	2016-03-17 14:53:36.805836853 +0100
@@ -332,7 +332,7 @@
             queue = gwbuf_make_contiguous(queue);
         }
 
-        if (query_classifier_get_operation(queue) & (QUERY_OP_DELETE | QUERY_OP_INSERT | QUERY_OP_UPDATE))
+	if (qc_get_operation(queue) & (QUERY_OP_DELETE | QUERY_OP_INSERT | QUERY_OP_UPDATE))
         {
             if ((sql = modutil_get_SQL(queue)) != NULL)
             {
