--- Makefile.orig	2012-06-13 12:15:50.000000000 +0300
+++ Makefile	2012-06-13 12:15:55.000000000 +0300
@@ -554,8 +554,8 @@
 		$(INSTALL_TOUCH) $(bin-prefix)/$(bin-dir)/$(NAME) 
 		$(INSTALL_BIN) $(NAME) $(bin-prefix)/$(bin-dir)
 		# install opensips menuconfig
-		$(INSTALL_TOUCH) $(bin-prefix)/$(bin-dir)/osipsconfig
-		$(INSTALL_BIN) menuconfig/configure $(bin-prefix)/$(bin-dir)/osipsconfig
+		#$(INSTALL_TOUCH) $(bin-prefix)/$(bin-dir)/osipsconfig
+		#$(INSTALL_BIN) menuconfig/configure $(bin-prefix)/$(bin-dir)/osipsconfig
 		# install opensipsctl (and family) tool
 		cat scripts/opensipsctl | \
 		sed -e "s#/usr/local/sbin#$(bin-target)#g" | \
