--- freeswitch-1.6.8/configure.ac	2016-05-05 20:42:20.000000000 +0000
+++ freeswitch-1.6.8.new/configure.ac	2016-05-25 19:34:00.823535317 +0000
@@ -414,15 +414,10 @@
 elif test "x${ax_cv_c_compiler_vendor}" = "xclang" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC)
-    APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
 elif test "x${ax_cv_c_compiler_vendor}" = "xgnu" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC)
     AC_SUBST([AM_MOD_AVMD_CXXFLAGS], [-std=gnu99])      # FS-8809, needed for MAP_POPULATE
-    if test "$ac_cv_gcc_supports_w_no_unused_result" = yes; then
-      APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
-      APR_ADDTO(SWITCH_AM_CFLAGS, -Wno-unused-result)
-    fi
     if test "$ac_cv_gcc_supports_w_no_misleading_indentation" = yes; then
       APR_ADDTO(SWITCH_AM_CFLAGS, -Wno-misleading-indentation)
     fi
