diff -Nur freeswitch-9999/src/mod/applications/mod_hash/Makefile.am freeswitch-src/src/mod/applications/mod_hash/Makefile.am
--- freeswitch-9999/src/mod/applications/mod_hash/Makefile.am	2014-12-24 07:39:54.864778202 +0800
+++ freeswitch-src/src/mod/applications/mod_hash/Makefile.am	2014-12-24 08:18:19.038697288 +0800
@@ -4,7 +4,7 @@
 ESL_DIR=$(switch_srcdir)/libs/esl
 
 mod_LTLIBRARIES = mod_hash.la
-mod_hash_la_SOURCES  = mod_hash.c ../../../../libs/esl/src/esl.c ../../../../libs/esl/src/esl_json.c ../../../../libs/esl/src/esl_event.c ../../../../libs/esl/src/esl_threadmutex.c ../../../../libs/esl/src/esl_config.c ../../../../libs/esl/src/esl_buffer.c
+mod_hash_la_SOURCES  = mod_hash.c 
 mod_hash_la_CFLAGS   = $(AM_CFLAGS) -I$(ESL_DIR)/src/include
 mod_hash_la_LIBADD   = $(switch_builddir)/libfreeswitch.la
 mod_hash_la_LDFLAGS  = -avoid-version -module -no-undefined -shared
