diff --git a/src/switch_core_media_bug.c b/src/switch_core_media_bug.c
index d6f7009..26c7230 100644
--- a/src/switch_core_media_bug.c
+++ b/src/switch_core_media_bug.c
@@ -778,6 +778,7 @@ SWITCH_DECLARE(switch_status_t) switch_core_media_bug_remove_all_function(switch
 	}

 	if (switch_core_codec_ready(&session->bug_codec)) {
+		PROTECT_INTERFACE(session->bug_codec.codec_interface);
 		switch_core_codec_destroy(&session->bug_codec);
 	}
